scm/mock.go

```diff
@@ -39,16 +39,21 @@ func NewMockSCMClient() *MockSCM {
 	return s
 }
 
-func (MockSCM) Clone(_ context.Context, opt *CloneOptions) (string, error) {
-	cloneDir := filepath.Join(opt.DestDir, repoDir(opt))
+func (s MockSCM) Clone(ctx context.Context, opt *CloneOptions) (string, error) {
```

Comment:
What I meant in #799 was that we should create and populate the `testdata` folder with some content and add it to GitHub, such as in `ci` or `assignments`, so that we could remove the hack below that creates a directory.

Reply:
That is, when mocking SCM, we could provide some "cloned" content that other tests can rely on.

Reply:
Right, my `scm/testdata` didn't get pushed to GitHub because the folders were empty, added some files now.